src/main/java/org/opensearch/knn/training/TrainingDataSpec.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.training;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * A data spec containing relevant training data for validation.
+ */
+@Getter
+@Setter
+@AllArgsConstructor
+public class TrainingDataSpec {
```

Comment:
Can we change this to DataSetInfo or VectorSpaceInfo? I think we might want to use this for non-training data as well. In the future, I could see it having the following properties:
1. dimension
2. vector count (either training or index)
3. space type

Reply:
Sure, I think I like the second one better